### PR TITLE
feat(urlt): replace internal/urlparser with pkg/urlt modified net/url parser

### PR DESCRIPTION
**Pull request by evg4b**
## Summary

Introduces `pkg/urlt` — a modified copy of Go's `net/url` stdlib — and replaces the old `internal/urlparser` package with it.

### Why

The previous `internal/urlparser` used a regex-based approach with `golang.org/x/net/idna` to parse proxy target URLs. It worked but had several limitations: it required masking `{placeholder}` tokens before parsing and un-masking afterwards, and it couldn't round-trip URLs containing placeholders without loss. The new package builds directly on the stdlib's own tokenizer, modified at the encoding-table level so `{` and `}` are treated as legal host characters — no masking tricks needed.

#### `pkg/urlt` (new)
- **`url.go`** — faithful `net/url`-style parser (modified copy of Go stdlib `net/url`) with a custom encoding table that keeps `{`/`}` unescaped in host position
- **`parser.go`** — opinionated, host-biased `Parse` / `ParseWithDefaultScheme` built on top:
  - Bare tokens like `example.com` or `localhost:8080` are resolved as **hosts**, not paths
  - Host is lowercased; only valid DNS / IPv4 / IPv6 / `{placeholder}` characters accepted
  - `{name}` placeholders parsed directly (e.g. `http://{client}.demo.com`)
- **`encoding_table.go`** — generated encoding table (via `go generate`)
- **`ToString`** — serializes a `*url.URL` back to string while preserving placeholder braces (stdlib `url.URL.String()` would percent-encode them)
- **`SplitHostPort`** — splits host/port from a parsed URL, keeping `[IPv6]` brackets intact

#### `internal/urlparser` (deleted)
The old regex+idna-based parser and its tests are removed. All callers updated to use `pkg/urlt`.

#### `main.go`
URL passwords are now redacted before being written to log output (`url.URL.Redacted()`), fixing a CodeQL CWE-312 "clear-text logging of sensitive information" alert.

## Test plan

- [ ] `make` passes (lint + test + build) — currently green
- [ ] SonarCloud Quality Gate: passed, 0 new issues
- [ ] CodeQL: passed
- [ ] `pkg/urlt` tests cover: bare hostnames, IPv4/IPv6, IDN, `{placeholder}` hosts, scheme normalisation, malformed inputs, `ToString` round-trips, `SplitHostPort` edge cases

### CODE REVIEW

File: pkg/urlt/parser_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseRequestURI(t *testing.T) {
+	tests := []struct {
+		in  string
+		out *url.URL
+		err bool
+	}{
+		// {in: "http://localhost", out: &url.URL{Scheme: "http", Host: "localhost"}},
```
Comment (evg4b) [inline pkg/urlt/parser_test.go]: All these test cases should be uncommeted

File: pkg/urlt/README.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+// Path segment escaping
+pathEscaped := urlt.PathEscape("path/to/resource")
+pathUnescaped, err := urlt.PathUnescape("path%2Fto%2Fresource")
+```
+
+### Query Parsing
+
+```go
+values, err := urlt.ParseQuery("name=John&age=30&hobby=reading")
```
Comment (evg4b) [inline pkg/urlt/README.md]: ParseQuery is the same like url.ParseQuery and it can be deleted

File: pkg/urlt/README.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+if err != nil {
+    log.Fatal(err)
+}
+```
+
+### Query and Path Escaping
+
+```go
+// Query parameter escaping
+escaped := urlt.QueryEscape("hello world?")  // "hello+world%3F"
```
Comment (evg4b) [inline pkg/urlt/README.md]: ParseQuery is the same like url.QueryEscape and it can be deleted

File: pkg/urlt/parser_test.go (hunk 1 above)
Comment (evg4b) [reply pkg/urlt/parser_test.go]: Deleted parser_test.go — its test cases were already fully covered by parse_test.go, which correctly calls urlt.Parse (the opinionated host-biased parser).

File: pkg/urlt/README.md (hunk 2 above)
Comment (evg4b) [reply pkg/urlt/README.md]: Removed ParseQuery from the package and deleted the Query Parsing section from README.md.

File: pkg/urlt/README.md (hunk 3 above)
Comment (evg4b) [reply pkg/urlt/README.md]: Removed QueryEscape, PathEscape, QueryUnescape, and PathUnescape from the package (all duplicates of net/url), and removed the Query and Path Escaping section from README.md.
